src/langchain_google_cloud_sql_pg/cloudsql_vectorstore.py

```diff
@@ -290,3 +297,53 @@ def similarity_search(
         self, query: str, k: int = 4, **kwargs: Any
     ) -> List[Document]:
         return []
+
+    async def aapply_vector_index(
+        self,
+        index: BaseIndex,
+        name: Optional[str] = None,
```

Comment:
Should this allow a Vector object as well?

Reply:
Can you add more? Since I have this on the vector store I don't think so?

Reply:
should be able to do this: (might need to use `Union[str, BaseIndex]` depends on when the other format was added python version wise) 
```suggestion
        name: Optional[str | BaseIndex] = None,
```

Reply:
We are already requiring a BaseIndex as the index argument, why does name need to support BaseIndex as well?